### PR TITLE
Translation

### PR DESCRIPTION
**Pull request by cristinagenduso**
I added the i18next library to manage the translation of the website, created the files for the translations (which, however, will have to be compiled over time). The file that has the most changes is that of the navbar where I have both added the changeLanguage function and added the first translations. I then made some minor changes. Among these:
- updated gitignore and prettierignore files to remove non-existent files/folders
- updated tailwind.config file to remove plugin section
- updated vite and tsconfig file to support import aliases (now you can use "@atoms", "@molecules", "@organisms", instead of having to specify "../../components/atoms...")
- created the 404 page
- updated html file with website title and changed index.jsx to index.tsx inside the script
- added first translations for the navbar

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const baseUrl = i18n.language === "it" ? "" : "/" + i18n.language
+
+  return (
+    <Suspense fallback="loading">
+      <Routes>
+        <Route element={<Layout />}>
+          <Route path={`/${baseUrl}`} element={<Home />} />
+          <Route path={`/${baseUrl}/about`} element={<About />} />
+          <Route path={`/${baseUrl}/projects`} element={<Projects />} />
+          <Route path={`/${baseUrl}/events`} element={<Events />} />
```
Comment (marc0777) [inline src/App.tsx]: The use of `baseUrl` in this way is very dangerous, and in this way would very likely break the current deployment (reminder for me that I should add more tests about that...).

At the very least before the `/$(baseUrl)` there should be a `.`, to ensure relative paths still works, 

A better way to handle this would be using [React Router's params](https://reactrouter.com/en/main/hooks/use-params). They used to support also optional parameters, so we could set a default language, however this is not the case [since v6](https://github.com/remix-run/react-router/issues/7285), but a couple of workarounds are given [here](https://stackoverflow.com/a/70961421/9226023).

File: src/components/atoms/NavBar/index.tsx [hunk 2]
```diff
@@ -23,22 +36,39 @@ const NavBar: FC<NavBarProps> = ({ setToggleMenu, toggleMenu }) => {
                 key={route.id}
                 className="relative inline-block mx-5 cursor-pointer h-fit"
               >
-                <NavLink to={route.path} className={navLinkClass}>
-                  {route.title}
+                <NavLink
+                  to={baseUrl + `/${route.path}`}
```
Comment (marc0777) [inline src/components/atoms/NavBar/index.tsx]: baseUrl here as well

File: src/components/organisms/Layout/index.types.ts [hunk 3]
```diff
@@ -1,3 +1,3 @@
 export default interface LayoutProps {
-  children: JSX.Element | JSX.Element[]
+  children?: JSX.Element | JSX.Element[]
```
Comment (marc0777) [inline src/components/organisms/Layout/index.types.ts]: Why the change?

File: package.json [hunk 4]
```diff
@@ -50,7 +52,7 @@
 		"@vitejs/plugin-react": "^4.0.0",
 		"eslint-config-prettier": "^8.8.0",
 		"eslint-plugin-prettier": "^4.2.1",
-		"eslint-plugin-react": "^7.0.1",
+		"eslint-plugin-react": "^7.32.2",
```
Comment (marc0777) [inline package.json:55]: Avoid updating dependencies in PRs that aren't strictly related

File: src/App.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import { Route, Routes } from "react-router-dom"
+import Home from "./pages"
+import NotFound from "./pages/NotFound"
+import About from "./pages/about"
+import Events from "./pages/events"
+import Projects from "./pages/projects"
+
+const App = () => {
+  const { i18n } = useTranslation()
+  const baseUrl = i18n.language === "it" ? "" : "/" + i18n.language
```
Comment (marc0777) [inline src/App.tsx]: Here an additional `/` is added, so when the language is not Italian the URL would end up being something like `example.org//en/about`. While this shouldn't create a problems per se, it's better to remove it

File: src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import ReactDOM from "react-dom/client"
+import { I18nextProvider } from "react-i18next"
+import { BrowserRouter } from "react-router-dom"
+import i18n from "../i18n"
+import App from "./App"
+import "./styles/global.css"
+
+ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
+  <React.StrictMode>
+    <BrowserRouter>
```
Comment (marc0777) [inline src/index.tsx:11]: I'm not convinced about the division: now `BrowserRouter` is split in half between here and `App.tsx` where the `Routes` are

**Review by marc0777 — COMMENTED** [review]
I added some comments. While not all of them are breaking changes, we should still evaluate them before merging.

Some edits here are connected to React Router, but maybe in the future we could consider using [vite-plugin-ssr](https://vite-plugin-ssr.com/), which integrates routing, support for i18n and many other things, while still being easily deployable on our environment. However I don't know if it's worth it or when we could do it.

Finally - I know I sound repetitive and I'm sorry - but in the future I suggest making PR's scope more narrow (e.g. only adding the translation and nothing else), as they are easier to review and it minimizes the number of bugs that could get through!

File: src/index.tsx (hunk 6 above)
Comment (cristinagenduso) [reply src/index.tsx:11]: In the documentation always appears a single `App` file with `Routes` and an index file with the `BrowserRouter`, also for the translation with i18n put everything in a single file gave me error. However, I couldn’t find any indication of the best practices to follow so I assumed this was the correct one

File: src/components/organisms/Layout/index.types.ts (hunk 3 above)
Comment (cristinagenduso) [reply src/components/organisms/Layout/index.types.ts]: the `children` is no longer necessary, but before I completely delete the `index.types.ts` file I wanted to do some testing with BrowserRouter to understand how to make the layout work best. Anyway, now I’ve removed it

File: src/App.tsx (hunk 5 above)
Comment (cristinagenduso) [reply src/App.tsx]: To avoid weighing down the code and especially maybe to have problems in the future, I decided to leave the universal url, operating only a change of language in the content, so that we are sure that everything will be successful. Later we can try again maybe to do something like this, however I don’t think there is a better solution, using only `react-router-dom` and not `next`

File: src/App.tsx (hunk 1 above)
Comment (cristinagenduso) [reply src/App.tsx]: From version 6.5 have been reintroduced, only, I don’t know why, it gave me problems with the home, which was not rendered in Italian, so I decided to try this solution that seemed to work

**Comment by cristinagenduso** [comment]
Thanks for the advice @marc0777 , I tried to fix it, let me know what you think
As for the PR, I forgot to push the branch and so all the changes accumulated together, although some were external to the topic of language change and therefore out of context, both with regard to the branch and the PR.

File: src/App.tsx (hunk 1 above)
Comment (marc0777) [reply src/App.tsx]: I admit I haven't actually tested if it worked or not in the case a different `BASE_PATH`, but by just reading the code it looks likely.

In case you too want to test in the future, you can do so by executing `set BASE_PATH=/isaac-website` before running `npm start dev`, that should deploy the website in the subfolder in Windows. To unset the variable `set BASE_PATH=`, or just close and reopen the terminal.

File: src/App.tsx (hunk 5 above)
Comment (marc0777) [reply src/App.tsx]: I think we should definitely look into [vite-plugin-ssr](https://vite-plugin-ssr.com/), as it seems a better [alternative to Next.js ](https://vite-plugin-ssr.com/nextjs-comparison) at least when using Vite, and also supports custom HTTP server (in our case Polito's ones) and not only the built in server that Next.js uses

Comment (cristinagenduso) [reply src/App.tsx]: Okay perfect, later I’ll take a look, because I’ve never heard of it. Maybe for now we can close this PR and come back to translations later, if you agree. In this way we avoid to make now this merge and in future maybe another change always linked to the translation

File: src/index.tsx (hunk 6 above)
Comment (marc0777) [reply src/index.tsx:11]: Do you have a link for the docs? I wasn't able to find it

Comment (cristinagenduso) [reply src/index.tsx:11]: Here you can directly find the code from the examples on [github](https://github.com/remix-run/react-router/tree/main/examples). For example, I looked at [this](https://github.com/remix-run/react-router/tree/main/examples/multi-app/home)

Comment (marc0777) [reply src/index.tsx:11]: Thanks! I was looking at `i18next` docs and not finding anything...😅 

The notation confuses me a bit, but they indeed use it everywhere so I guess there must be a reason
